diff --git a/libfdt/meson.build b/libfdt/meson.build
index bf8343f..0bc0781 100644
--- a/libfdt/meson.build
+++ b/libfdt/meson.build
@@ -26,19 +26,14 @@ else
 endif
 
 link_args += version_script
-libfdt = both_libraries(
+libfdt = static_library(
   'fdt', sources,
-  version: meson.project_version(),
   link_args: link_args,
   link_depends: 'version.lds',
   install: true,
 )
 
-if static_build
-  link_with = libfdt.get_static_lib()
-else
-  link_with = libfdt.get_shared_lib()
-endif
+link_with = libfdt
 
 libfdt_inc = include_directories('.')
 
